src/client/src/ui/common/popout/popoutServiceFactory.js

```diff
@@ -0,0 +1,13 @@
+import BrowserPopoutService from './browserPopoutService';
+import OpenfinPopoutService from './openfinPopoutService';
+
+
+export default class PopoutServiceFactory {
+  constructor(openfin) {
+    this._openfin = openfin;
+  }
+
+  getService() {
+    return window.fin ? new OpenfinPopoutService(this._openfin) : new BrowserPopoutService();
```

Comment:
there is a prop on `OpenFin` called `isRunningInOpenFin`  we should point to that for consistency.